app/controllers/EditionsController.scala

```diff
@@ -110,17 +130,28 @@ class EditionsController(db: EditionsDB,
   }
 
   def listIssues(edition: Edition) = EditEditionsAuthAction { req =>
-    Try {
+    val params = Try {
       val dateFrom = req.queryString.get("dateFrom").map(_.head).get
       val dateTo = req.queryString.get("dateTo").map(_.head).get
       (LocalDate.parse(dateFrom), LocalDate.parse(dateTo))
-    }.map {
-      case (localDateFrom, localDateTo) =>
-        db.listIssues(edition, localDateFrom, localDateTo) match {
-          case Right(issues) => Ok(Json.toJson(issues))
-          case Left(errors) => InternalServerError(s"Error listing issues: ${errors.mkString(", ")}")
-        }
-    }.getOrElse(BadRequest("Invalid or missing date"))
+    }.toEither.left.map { _ =>
+      BadRequest("Invalid or missing date")
```

Comment:
Should we include the details of the error here?

Reply:
Yes, that's nicer! e95e5d86e7b9c9385fc79b0214fb9deade80bb61